framework/Optimizers/gradients/SPSA.py

```diff
@@ -0,0 +1,19 @@
+import abc
+import copy
+import pandas as pd
+from utils import InputData, InputTypes, randomUtils, mathUtils
+
+from utils import InputData, InputTypes, randomUtils, mathUtils
+
+from .GradientApproximater import GradientApproximater
+
+from .FiniteDifference import FiniteDifference
+
+class Stochastic(FiniteDifference):
```

Comment:
As we discussed, let's keep central difference and SPSA as part of separate merge requests, and SPSA should overwrite all the same methods as central diff, so both should inherit separately from GradientApproximater base class.

Reply:
Alright!